Mathlib/Algebra/GroupWithZero/WithZero.lean

```diff
@@ -314,15 +314,26 @@ end WithZero
 
 namespace MonoidWithZeroHom
 
-protected lemma map_eq_zero_iff {G₀ G₀' : Type*} [GroupWithZero G₀]
-    [MulZeroOneClass G₀'] [Nontrivial G₀']
-    {f : G₀ →*₀ G₀'} {x : G₀}:
+variable {G₀' : Type*} [GroupWithZero G₀']
+
+/-- The map `withZeroUnitsEquiv` as a `MonoidWithZeroHom`. -/
+def withZeroUnitsHom [DecidablePred (fun a : G₀' ↦ a = 0)] :
+    WithZero G₀'ˣ →*₀ G₀' where
+  __ := WithZero.withZeroUnitsEquiv
+  map_zero' := rfl
+  map_one' := rfl
+
+lemma withZeroUnitsHom_inj [DecidablePred (fun a : G₀' ↦ a = 0)] :
```

Comment:
```suggestion
lemma withZeroUnitsHom_injective [DecidablePred (fun a : G₀' ↦ a = 0)] :
```

Reply:
We have tons of `_inj` lemmas, and the name is sufficiently long already... or are the `_inj` and `_injective`  convention different?